compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.cpp

```diff
@@ -113,6 +113,20 @@ SmallVector<int64_t> TilingConfig::getFusableLevels() {
   }
 }
 
+SmallVector<int64_t> TilingConfig::getCacheLevels() {
+  switch (getNumTilingLevels()) {
+  case 0:
+  case 1:
+  case 4:
+    return {};
+  case 6:
+    // Cache parallel and cache reduction levels.
+    return {1, 2};
+  default:
+    llvm_unreachable("Unexpected number of tiling levels");
```

Comment:
it seems hard to guarantee that this will always be unreachable, and this isn't so performance-critical code that it would be worth the risk. Downgrade this to either `assert` or error propagation.